# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieNotification.h
# Commit: 5aa3b2f1b5ec
# Full Hash: 5aa3b2f1b5ec8a63fa04f80cf4aed4aac2c6ce22
# Author: Paul Zuehlcke <pbz@mozilla.com>
# Date: 2023-09-07 21:34:02
# Regressor Bug: 1839918
# File Overlap Count: 6
# Description:
#   Bug 1839918 - Refactor "cookie-changed" notifications to use nsICookieNotification. r=extension-reviewers,cookie-reviewers,sessionstore-reviewers,bvandersloot,edgul,robwu
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D185300
# ==============================================================================

diff -r fecbf97cc8db -r 5aa3b2f1b5ec netwerk/cookie/CookieNotification.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/netwerk/cookie/CookieNotification.h	Thu Sep 07 08:25:54 2023 +0000
@@ -0,0 +1,49 @@
+/* -*- Mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef mozilla_net_CookieNotification_h
+#define mozilla_net_CookieNotification_h
+
+#include "nsIArray.h"
+#include "nsICookieNotification.h"
+#include "nsICookie.h"
+#include "nsCOMPtr.h"
+#include "nsTArray.h"
+#include "nsString.h"
+
+namespace mozilla::net {
+
+class CookieNotification final : public nsICookieNotification {
+ public:
+  // nsISupports
+  NS_DECL_ISUPPORTS
+  NS_DECL_NSICOOKIENOTIFICATION
+
+  explicit CookieNotification(nsICookieNotification::Action aAction,
+                              nsICookie* aCookie, const nsACString& aBaseDomain,
+                              nsIArray* aBatchDeletedCookies = nullptr,
+                              uint64_t aBrowsingContextId = 0,
+                              bool aIsThirdPartyCookie = false)
+      : mAction(aAction),
+        mCookie(aCookie),
+        mBaseDomain(aBaseDomain),
+        mBatchDeletedCookies(aBatchDeletedCookies),
+        mBrowsingContextId(aBrowsingContextId),
+        mIsThirdPartyCookie(aIsThirdPartyCookie){};
+
+ private:
+  nsICookieNotification::Action mAction;
+  nsCOMPtr<nsICookie> mCookie;
+  nsCString mBaseDomain;
+  nsCOMPtr<nsIArray> mBatchDeletedCookies;
+  uint64_t mBrowsingContextId = 0;
+  bool mIsThirdPartyCookie = false;
+
+  ~CookieNotification() = default;
+};
+
+}  // namespace mozilla::net
+
+#endif  // mozilla_net_CookieNotification_h